enhanced/scraper.go

```diff
@@ -59,60 +59,72 @@ func (s *scraper) start(ctx context.Context, interval time.Duration, ch chan<- m
+
+		collectAllMetrics := func(output *cloudwatchlogs.FilterLogEventsOutput, lastPage bool) bool {
+			for _, event := range output.Events {
+				l := s.logger.With("EventId", *event.EventId).With("LogStreamName", *event.LogStreamName)
+				l = l.With("Timestamp", aws.MillisecondsTimeValue(event.Timestamp).UTC())
+				l = l.With("IngestionTime", aws.MillisecondsTimeValue(event.IngestionTime).UTC())
+
+				var instance *sessions.Instance
+				for _, i := range s.instances {
+					if i.ResourceID == *event.LogStreamName {
+						instance = &i
+						break
+					}
 				}
-			}
-			if instance == nil {
-				l.Errorf("Failed to find instance.")
-				continue
-			}
-			l = l.With("region", instance.Region).With("instance", instance.Instance)
+				if instance == nil {
+					l.Errorf("Failed to find instance.")
+					continue
+				}
+				l = l.With("region", instance.Region).With("instance", instance.Instance)
 
-			// l.Debugf("Message:\n%s", *event.Message)
-			osMetrics, err := parseOSMetrics([]byte(*event.Message))
-			if err != nil {
-				l.Errorf("Failed to parse metrics: %s.", err)
-				continue
-			}
-			// l.Debugf("OS Metrics:\n%#v", osMetrics)
+				// l.Debugf("Message:\n%s", *event.Message)
+				osMetrics, err := parseOSMetrics([]byte(*event.Message))
+				if err != nil {
+					l.Errorf("Failed to parse metrics: %s.", err)
+					continue
+				}
+				// l.Debugf("OS Metrics:\n%#v", osMetrics)
 
-			if allMetrics[instance.ResourceID] == nil {
-				allMetrics[instance.ResourceID] = make(map[time.Time][]prometheus.Metric)
+				if allMetrics[instance.ResourceID] == nil {
+					allMetrics[instance.ResourceID] = make(map[time.Time][]prometheus.Metric)
+				}
+				timestamp := aws.MillisecondsTimeValue(event.Timestamp)
+				metrics := osMetrics.makePrometheusMetrics(instance.Region)
+				allMetrics[instance.ResourceID][timestamp] = metrics
+				l.Debugf("Timestamp from Message: %s.", osMetrics.Timestamp.UTC())
 			}
-			timestamp := aws.MillisecondsTimeValue(event.Timestamp)
-			metrics := osMetrics.makePrometheusMetrics(instance.Region)
-			allMetrics[instance.ResourceID][timestamp] = metrics
-			l.Debugf("Timestamp from Message: %s.", osMetrics.Timestamp.UTC())
+
+			return true // continue pagination
 		}
 
-		return true // continue pagination
-	}
-	if err := s.svc.FilterLogEventsPagesWithContext(ctx, input, collectAllMetrics); err != nil {
-		if awsErr, ok := err.(awserr.Error); ok {
-			if awsErr.Code() == cloudwatchlogs.ErrCodeResourceNotFoundException {
-				s.logger.Warnf("Enhanced monitoring problem: %s", awsErr.Message())
+		if err := s.svc.FilterLogEventsPagesWithContext(ctx, input, collectAllMetrics); err != nil {
+			if awsErr, ok := err.(awserr.Error); ok {
+				if awsErr.Code() == cloudwatchlogs.ErrCodeResourceNotFoundException {
+					s.logger.Warnf("Enhanced monitoring problem: %s", awsErr.Message())
+				}
+			} else {
+				s.logger.Errorf("Failed to filter log events: %s.", err)
```

Comment:
Seems like the period at the end is there in the original, but Golang prefers log messages to not end with punctuation (they might be expanded on or added to, so using punctuation is not recommended at their end).